kornia/geometry/keypoints.py

```diff
@@ -135,15 +167,17 @@ def transform_keypoints(self, M: Tensor, inplace: bool = False) -> "Keypoints":
             raise ValueError(f"The transformation matrix shape must be (3, 3) or (B, 3, 3). Got {M.shape}.")
 
         transformed_boxes = transform_points(M, self._data)
+
         if inplace:
             self._data = transformed_boxes
+            self._update_validity_mask(params)
             return self
 
-        return Keypoints(transformed_boxes, False)
+        return Keypoints(transformed_boxes, False, self._image_size, self._visibility)
 
-    def transform_keypoints_(self, M: Tensor) -> "Keypoints":
+    def transform_keypoints_(self, M: Tensor, params: Dict[str, Tensor]) -> "Keypoints":
```

Comment:
I suggest using `image_size` here, instead of `params`.

Reply:
The issue here is that there are multiple [from what I can tell, 3] calls to this in `kornia/augmentation/_2d/geometric/base.py` where we'd need to specify the `image_size` logic now in `update_valid_mask`.

Reply:
i agree with @shijianjian, keypoint shouldn't be aware of any dictionary logic and just receive image size (we even have a data structure for this). Keypoint should be as light as possible and any extra logic should be part of the augmentation package to keep the whole thing more maintainable.

Reply:
> The issue here is that there are multiple [from what I can tell, 3] calls to this in `kornia/augmentation/_2d/geometric/base.py` where we'd need to specify the `image_size` logic now in `update_valid_mask`.

We may have an internal function inside augmentation modules to update `image_size`, where `update_validity_mask` can be called.

Reply:
Could you maybe point me to it? The only thing I could find is `_get_new_batch_shape` [here](https://github.com/alexanderswerdlow/kornia/blob/92deb42b56468ca36392f53d2a6b3616ca9e169d/kornia/augmentation/container/image.py#L290) which is called from `forward_parameters` and `forward` in `ImageSequential`.

Perhaps we put the dictionary logic [here](https://github.com/alexanderswerdlow/kornia/blob/92deb42b56468ca36392f53d2a6b3616ca9e169d/kornia/augmentation/base.py#L345) in `transform_keypoints`? It seems like we'd need to update all of the `apply_transform_keypoint` definitions however.